components/budget/filters/PeriodFilter.js

```diff
@@ -48,10 +48,16 @@ const getIntervalFromValue = value => {
  */
 const getNewInterval = (interval, changeField, newValue) => {
   const newInterval = { ...interval };
-  newInterval[changeField] = stripTime(newValue);
+  newInterval[changeField] = newValue;
 
   // Reset interval in case fromDate is after toDate
-  if (newInterval.from && newInterval.to && newInterval.from > newInterval.to) {
+  if (
+    newInterval.from &&
+    !newInterval.from.startsWith('0') &&
+    newInterval.to &&
+    !newInterval.to.startsWith('0') &&
+    newInterval.from > newInterval.to
+  ) {
```

Comment:
That works for me, but let's add some clues to help other developers understand what's going on here. We could add some comments, or extract that code to a clearly named helper:

```suggestion
  if (isValidDate(newInterval.from) && isValidDate(newInterval.to) && newInterval.from > newInterval.to) {
```

Reply:
Thanks @Betree  I have made the change